flang/lib/Optimizer/Transforms/ArrayValueCopy.cpp

```diff
@@ -420,27 +420,41 @@ void ArrayCopyAnalysis::arrayMentions(
   loadMapSets.insert({load, visited});
 }
 
+static bool hasPointerAttribute(mlir::Type type) {
+  if (auto boxTy = type.dyn_cast<BoxType>())
+    type = boxTy.getEleTy();
+  return type.isa<fir::PointerType>();
+}
+
 /// Is there a conflict between the array value that was updated and to be
 /// stored to `st` and the set of arrays loaded (`reach`) and used to compute
 /// the updated value?
 static bool conflictOnLoad(llvm::ArrayRef<mlir::Operation *> reach,
                            ArrayMergeStoreOp st) {
   mlir::Value load;
-  auto addr = st.memref();
+  bool storeMemRefWasSeen = false;
```

Comment:
Why this change? There is no order implied in the `reach` set, so we'll not guaranteed to find the array_load corresponding to the array_merge_store first. So the name of this variable is misleading.

The original algorithm keeps track of the first array_load found in the list and short-circuits if it finds more than one. A logical extension of that is to do a better analysis on the array_load ops that are found. Using a `bool` doesn't get there.

Reply:
Thanks for explaining, I reverted that change.